src/libOpenImageIO/imagebufalgo_test.cpp

```diff
@@ -1264,6 +1264,59 @@ test_simple_perpixel()
 
 
 
+static void
+test_demosaic()
+{
+    print("Testing demosaic\n");
+
+    std::string inputs[] = { "/Users/ad/Desktop/TestRaw/Image1.NEF",
+                             "/Users/ad/Desktop/TestRaw/Image2.DNG",
+                             "/Users/ad/Desktop/TestRaw/Image3.RW2" };
```

Comment:
This is obviously not going to work for anybody but you. :-)

Not every IBA function needs to be tested (or can be done in a non-awkward way) in imagebufalgo_test.cpp. That's a decent place to put a self-contained unit test if you can programmatically construct an ImageBuf containing a low-resolution synthetic input designed so that you can inspect its output to be sure it generates the numerically correct results. As you can see from most of the rest of that file.

In general, the more robust tests on real images tend to be the result of implementing the Python bindings and exposing the IBA function as an oiiotool command (both of which will be necessary anyway, as direct use of the C++ APIs are a tiny sliver of typical OIIO use), and then it's much easier to construct more complex python or oiiotool command line based tests on real data (which if they are very big can be stored in the OpenImageIO-images project so as not to bulk up the main source code repo unnecessarily).